modules/tensor_mechanics/src/materials/LAROMANCEStressUpdateBase.C

```diff
@@ -856,21 +892,25 @@ ADLAROMANCEStressUpdateBase::computeTimeStepLimit()
   return limited_dt;
 }
 
-ADReal
-ADLAROMANCEStressUpdateBase::sigmoid(const Real lower,
-                                     const Real upper,
-                                     const ADReal & val,
-                                     const bool derivative)
+template <bool is_ad>
+GenericReal<is_ad>
+LAROMANCEStressUpdateBaseTempl<is_ad>::sigmoid(const Real lower,
+                                               const Real upper,
+                                               const GenericReal<is_ad> & val,
+                                               const bool derivative)
 {
-  ADReal x = (val - lower) / (upper - lower);
+  GenericReal<is_ad> x = (val - lower) / (upper - lower);
   x = 2.0 * x - 1.0;
-  ADReal plus = std::exp(-2.0 / (1.0 + x));
-  ADReal minus = std::exp(-2.0 / (1.0 - x));
+  GenericReal<is_ad> plus = std::exp(-2.0 / (1.0 + x));
```

Comment:
you should use more `auto`!!

Reply:
> The `auto` comment only affects maybe a dozen lines. This can be fixed later

Can GenericReal<is_ad> be replaced with auto?

Reply:
Yes, as long at the type can be deduced from the right hand side. This wont work if the RHS is a floating point literal, but it will work if the right hand side is an expression containing `GenericReal<is_ad>` typed variables.